### PR TITLE
Added a changelog

### LINKED ISSUE #3941: Does ISIS have a "history" file?
**Description**
Many open source projects have a HISTORY file in the main level of the repository that provides a high-level overview of features and bug fixes in each release.  There appears to be no such file in the ISIS3 repo.  Maybe it is elsewhere in the hierarchy?

There does appear to be a [thread on AstroDiscuss](https://astrodiscuss.usgs.gov/c/ISIS/isis-release-notes) which contains this information, but it is separate from the repo, which seems odd.  It is easy enough to PR a HISTORY file into existence based on that thread, but I wondered if maybe there was something I was missing.

**Suggested Edit**
Create and maintain a HISTORY file in the main level of the repo.

Thanks!

### PR DESCRIPTION
**Pull request by jessemapel**
## Description

I added a changelog based on https://keepachangelog.com/en/1.0.0/ and populated it with the changelogs from all the releases that we have announcements for on astrodiscuss. I also updated the PR template to remind folks to add their change to the changelog if they impact users.

I also made a small change to install the AUTHORS file.

**note:** There will need to be a follow-on update to the release process to cover the new changelog policy. I'll do this once this PR is merged.

## Related Issue

Fixes #3941

## Motivation and Context

Currently generate changelogs using github diffs and then post them on astrodiscuss. This PR address two problems with that.

First, the person pulling the release has to write the changelog and may not be familiar with all of the changes in that release. By having the contributor who submitted the PR write their changelog entry they will likely be of a higher quality. It also submits them to a peer review process.

Second, the changelog is kept with the code base and distributed in the isntallation. This way contributors to the repo do not need to go to another, user focused sight. Also, users have a local copy of the changelog that lists more than just the latest release.

## How Has This Been Tested?

I rendered all of the markdown on my branch and did not find anything that caused errors. Some of the links will not work until this PR has been merged, but they should be staged appropriately so that nothing needs to be done post merge.

I also built and installed the code to ensure the files installed correctly.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have added myself to the [AUTHORS.rst](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/AUTHORS.rst) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by rbeyer — DISMISSED** [review]

**Review by jlaura — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+ - caminfo now properly errors when passed a projected cube.
+ - Filenames with months in them are now properly translated using the host system's local instead of US English.
+ - findfeatures now properly resets the input images between algorithm runs when running in multi-algorithm mode.
+ - qmos no longer hangs when drawing a grid over a projected image with positive west longitude domain.
+
+## [3.8.0] - 2019-07-29
+
+### Changed
+
+ - Removed some dev tools form the installation environment.
```
Comment (krlberry) [inline CHANGELOG.md]: I'm sure this came from the original astrodiscuss post, but can you update form -> from?

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+### Fixed
+
+ - Fixed ingestion of Rosetta VIRITIS lvl2 spectra.
+
+## [3.7.1] - 2019-05-30
+
+### Fixed
+
+- Paths are no longer break findfeature's algorithm parameter.
```
Comment (krlberry) [inline CHANGELOG.md]: Similarly, remove "are" from the above?

**Review by krlberry — CHANGES_REQUESTED** [review]
This looks great! Just found a couple of small typos (probably from the original astrodiscuss posts) that might as well get fixed now.

**Review by krlberry — APPROVED** [review]
Thanks!

**Comment by jessemapel** [comment]
Going to wait for #3992 to merge because they will likely have a merge conflict.

**Comment by krlberry** [comment]
@jessemapel I don't see a conflict. Good to merge?

**Pull request merged**